lib/auth/plain-text-auth-provider.js

```diff
@@ -19,50 +19,48 @@ const Authenticator = provider.Authenticator;
  * @alias module:auth~PlainTextAuthProvider
  * @constructor
  */
-function PlainTextAuthProvider(username, password) {
-    this.username = username;
-    this.password = password;
+class PlainTextAuthProvider {
+    constructor(username, password) {
+        this.username = username;
+        this.password = password;
+    }
+    /**
+     * Returns a new [Authenticator]{@link module:auth~Authenticator} instance to be used for plain text authentication.
+     * @override
+     * @returns {Authenticator}
+     */
+    newAuthenticator() {
+        return new PlainTextAuthenticator(this.username, this.password);
+    }
 }
 
 util.inherits(PlainTextAuthProvider, AuthProvider);
```

Comment:
:question: Shouldn't this also be converted to the classes syntax, i.e., `extends`?